binding.gyp

```diff
@@ -25,22 +17,58 @@
       'dependencies': ["<!(node -p \"require('node-addon-api').gyp\")"],
       'conditions': [
         ['OS=="linux"', {
-          "libraries": [ "-L<!(pwd)/libpg_query/linux", "-lpg_query" ]
+          "libraries": [ "-L<!(pwd)/libpg_query/linux", "-lpg_query" ],
+          "actions": [
+            {
+              "outputs": ['libpg_query/include/pg_query.h'],
+              "inputs": [],
+              "action": ['script/buildAddon.sh'],
+              "action_name": 'prebuild_dependencies'
+            }
+          ],
         }],
         ['OS=="mac"', {
           "libraries": [ "-L<!(pwd)/libpg_query/osx", "-lpg_query" ],
           "xcode_settings": {
             "CLANG_CXX_LIBRARY": "libc++",
             'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
             'MACOSX_DEPLOYMENT_TARGET': '10.7'
-          }
+          },
+          "actions": [
+            {
+              "outputs": ['libpg_query/include/pg_query.h'],
+              "inputs": [],
+              "action": ['script/buildAddon.sh'],
+              "action_name": 'prebuild_dependencies'
+            }
+          ],
         }],
         ['OS=="win"', {
+          "link_settings": {
+            "library_dirs": [
+              "../libpg_query/windows"
+            ],
+            "libraries": [
+              "../libpg_query/windows/pg_query.lib"
+            ],
+          },
           "msvs_settings": {
             "VCCLCompilerTool": {
-              "ExceptionHandling": 1
+              "ExceptionHandling": 0,
+              "AdditionalOptions": ["/EHsc"]
+            }
+          },
+          "defines": [ "NAPI_DISABLE_CPP_EXCEPTIONS" ],
+          "actions":[
+            {
+              "outputs": [
+                ""
```

Comment:
this `outputs` field contains an empty string.
because empty content in `outputs` will let node-gyp skip the actions.

so it's a workaround keeping actions works.